scheduler/test/cook/test/compute_cluster.clj

```diff
@@ -0,0 +1,956 @@
+                                  :base-path "both2-base-path"}
+                         "both3" {:name "both3"
+                                  :a :b
+                                  :ca-cert 4
+                                  :base-path "both3-base-path"}
+                         "both4" {:name "both4"
+                                  :a :b
+                                  :ca-cert 5
+                                  :base-path "both4-base-path"}
+                         "both5" {:name "both5"
+                                  :a :b
+                                  :ca-cert 6
+                                  :base-path "both5-base-path"}
+
+                         "base-path-collide-existing"
+                         {:name "base-path-collide-existing"
+                          :ca-cert 7
+                          :state :deleted
+                          :base-path "path-collide"}
+
+                         }
+                        {"both1" {:name "both1"
+                                  :a :a
+                                  :ca-cert 2
+                                  :base-path "both1-base-path"}
+                         "both2" {:name "both2"
+                                  :a :b
+                                  :ca-cert 3
+                                  :base-path "both2-base-path"}
+                         "both3" {:name "both3"
+                                  :a :b
+                                  :ca-cert 4
+                                  :base-path "both4-base-path"}
+                         "both4" {:name "both4"
+                                  :a :b
+                                  :ca-cert 5
+                                  :base-path "both3-base-path"}
+                         "both5" {:name "both5"
+                                  :a :b
+                                  :ca-cert 6
+                                  :base-path "both5-base-path"}
+                         "right" {:name "right"
+                                  :a :a
+                                  :ca-cert 12
+                                  :base-path "right-base-path"}
+                         "right2" {:name "right2"
+                                   :a :a
+                                   :ca-cert 13
+                                   :base-path "both5-base-path"}
+
+                         "base-path-collide-existing"
+                         {:name "base-path-collide-existing"
+                          :ca-cert 7
+                          :state :deleted
+                          :base-path "path-collide"}
+                         "base-path-collide-new"
+                         {:name "base-path-collide-new"
+                          :ca-cert 7
+                          :base-path "path-collide"}
+
+                         }
+                        nil))]
+      (is (= expected actual)))))
+
+(def initialize-cluster-fn-invocations-atom (atom []))
```

Comment:
Not sure I like this global state. E.g., it could interfere with parallel unit test invocation. @dposada any thought of something better here?

Reply:
I think we do something similar elsewhere. open to suggestions for recording invocations

Reply:
See if @dposada  has any ideas. I played with a couple of solutions, but couldn't make them work.

Reply:
I think you could make `initialize-cluster` call a namespace-level function, and then `with-redefs` that function in each unit test, which would let you keep the `atom` local to each test.